File: src/main/java/me/lokka30/levelledmobs/nametag/NametagSender.java
package me.lokka30.levelledmobs.nametag;

import me.lokka30.levelledmobs.result.NametagResult;
import org.bukkit.entity.LivingEntity;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;

/**
 * A common interface for sending nametag packets
 *
 * @author PenalBuffalo (aka stumper66)
 * @since 3.6.0
 */
public interface NametagSender {

    void sendNametag(
        final @NotNull LivingEntity livingEntity,
        final @NotNull NametagResult nametag,
        final @NotNull Player player,
        final boolean alwaysVisible
    );
}

File: levelledmobs-plugin/src/main/kotlin/io/github/arcaneplugins/levelledmobs/nametag/NametagSender.kt
package io.github.arcaneplugins.levelledmobs.nametag

import io.github.arcaneplugins.levelledmobs.result.NametagResult
import org.bukkit.entity.LivingEntity
import org.bukkit.entity.Player

/**
 * A common interface for sending nametag packets
 *
 * @author PenalBuffalo (aka stumper66)
 * @since 3.6.0
 */
interface NametagSender {
    fun sendNametag(
        livingEntity: LivingEntity,
        nametag: NametagResult,
        player: Player,
        alwaysVisible: Boolean
    )
}